lib/internal/errors.js

```diff
@@ -828,16 +765,20 @@ function setArrowMessage(err, arrowMessage) {
 // Hide stack lines before the first user code line.
 function hideInternalStackFrames(error) {
   overrideStackTrace.set(error, (error, stackFrames) => {
-    let frames = stackFrames;
+    let result = '';
     if (typeof stackFrames === 'object') {
-      frames = ArrayPrototypeFilter(
-        stackFrames,
-        (frm) => !StringPrototypeStartsWith(frm.getFileName() || '',
-                                            'node:internal'),
-      );
+      for (const frame of stackFrames) {
+        const filename = frame.getFileName();
+        if (!filename || !StringPrototypeStartsWith(filename, 'node:internal')) {
+          result += `\n    at ${frame}`;
+        }
+      }
+    } else {
+      for (const frame of stackFrames) {
```

Comment:
for-of is vulnerable to Array.prototype Symbol.iterator modification, this should be a regular for loop, or ArrayPrototypeForEach

Reply:
Why would you prefer this over the much faster and more robust `for (let i = 0; i < stackFrames.length; i++) { const frame = stackFrames[i];`, in a performance improvement PR?

Reply:
Last time I checked, classic `for` loops are still the most efficient way of iterating over an array, I agree with Jordan, it'd be nice to use those here.

Reply:
@JakobJingleheimer it's also an issue of robustness. I don't care about while vs for - personal preference can rule there - but for-of is simply not safe like the others.